account_manager/src/validator/deposit.rs

```diff
@@ -79,48 +91,17 @@ pub fn cli_app<'a, 'b>() -> App<'a, 'b> {
         )
 }
 
-pub fn cli_run<T: EthSpec>(
-    matches: &ArgMatches<'_>,
-    mut env: Environment<T>,
-) -> Result<(), String> {
+fn sendDepositTransactions<T1, T2>(
+    mut env: Environment<T1>,
+    validators: Vec<validator_dir::ValidatorDir>,
+    from_address: Address,
+    transport: Box<dyn Transport>,
+) -> Result<(), String>
+where
+    T1: EthSpec,
```

Comment:
Hey @ethDreamer , you don't really need trait objects here imo. You can use simple generics to define something like

```rust
async fn do_deposits<T>(transport: T) -> Result<(), String>
where
    T: Transport,
{
    let web3 = Web3::new(transport);
    poll_until_synced(...).await;
    ....
}
```

Reply:
Hi @pawanjay176!
You're right! I didn't need trait objects. I was required to use a very strange syntax that I'm completely unfamiliar with on L103 to solve the compiler errors though.. would be helpful if anyone could point me to a page on that.

Reply:
So the [transport trait](https://docs.rs/web3/0.11.0/web3/trait.Transport.html#associated-types) has an associated type `Out`. The bound you added basically says that the `Out` type for whatever transport you pass should also implement the `Send` trait. 

The compiler forced you to add it because the `poll_until_synced` function you called also requires that bound (although I'm not sure why). 
[This](https://doc.rust-lang.org/book/ch19-03-advanced-traits.html#fully-qualified-syntax-for-disambiguation-calling-methods-with-the-same-name) might be a good starting point to understand it in more detail :)